### ISSUE TITLE
[Bug]: aws_vpc deletion waiting on IPAM pool deallocation

### ISSUE THREAD

**Issue by TibbeVandenBerghe** [hashicorp/terraform-provider-aws#31211]
### Terraform Core Version

1.3.9, 1.4.2

### AWS Provider Version

4.66.0, 4.64.0

### Affected Resource(s)

aws_vpc

### Expected Behavior

VPC will be deleted and aws itself will remove the allocation in the ipam pool. The deletion of the aws_vpc resource won't be waiting on the deallocation. Should take around 10-20seconds

### Actual Behavior

Terraform keeps waiting on the deletion of the aws_vpc resource. The VPC itself is already deleted on aws. The deletion is complete after the ipam allocation of the cidr from the vpc is deallocated. This takes 10-20minutes. Previously a vpc deletion took 10 seconds. This takes a lot of build minutes in the pipeline. Previously this wasn't an issue

### Terraform Configuration Files
```
resource "aws_vpc" "vpc" {
  ipv4_ipam_pool_id    = var.ipam_pool
  ipv4_netmask_length  = var.vpc_netmask_lenght
  enable_dns_support   = true
  enable_dns_hostnames = true
  tags = {
    Name = "${var.group_name}-vpc"
  }
}

variable "vpc_netmask_lenght" {
  description = "Lenght of the subnet in your vpc, please don't take more IPs then /24"
  type        = number
}

variable "ipam_pool" {
  description = "ID of ipam pool used to create the vpc cidr blocks"
  type = string
}
```
### Steps to Reproduce

Have an IPAM pool in aws and set it in the aws_vpc resource. 

- Plan
- Apply
- Destroy
---
If you cancel the destroy when it's stuck on the deletion of the aws_vpc resource it throws the error
"waiting on ipam allocation removal"
When you then run the destroy again it says no resources needed to destroy since the vpc is already destroyed but it just keeps waiting for the ipam deallocation.

### Debug Output

Debug during deletion:
```
module.production.module.vpc.aws_vpc.vpc: Still destroying... [id=vpc-****, 13m30s elapsed]
2023-05-05T09:41:26.224+0200 [DEBUG] provider.terraform-provider-aws_v4.64.0_x5: HTTP Request Sent: aws.region=us-east-2 http.flavor=1.1 http.request_content_length=87 http.url=https://ec2.us-east-2.amazonaws.com/ tf_provider_addr=registry.terraform.io/hashicorp/aws tf_rpc=ApplyResourceChange aws.operation=GetIpamPoolAllocations aws.sdk=aws-sdk-go aws.service=EC2 http.request.body=Action=GetIpamPoolAllocations&IpamPoolId=ipam-pool-****&Version=2016-11-15 http.request.header.authorization="AWS4-HMAC-SHA256 Credential=AKIA************BI2I/20230505/us-east-2/ec2/aws4_request, SignedHeaders=content-length;content-type;host;x-amz-date, Signature=*****" @module=aws tf_mux_provider=*schema.GRPCProviderServer tf_req_id=3c02edf2-d892-1132-26f2-dbb1083f9dbe tf_resource_type=aws_vpc @caller=github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2@v2.0.0-beta.26/logger.go:90 http.method=POST http.request.header.content_type="application/x-www-form-urlencoded; charset=utf-8" http.request.header.x_amz_date=20230505T074126Z http.user_agent="APN/1.0 HashiCorp/1.0 Terraform/1.3.9 (+https://www.terraform.io) terraform-provider-aws/4.64.0 (+https://registry.terraform.io/providers/hashicorp/aws) aws-sdk-go/1.44.246 (go1.19.8; linux; amd64)" net.peer.name=ec2.us-east-2.amazonaws.com timestamp=2023-05-05T09:41:26.224+0200
2023-05-05T09:41:26.557+0200 [DEBUG] provider.terraform-provider-aws_v4.64.0_x5: HTTP Response Received: aws.sdk=aws-sdk-go http.response.body="<?xml version="1.0" encoding="UTF-8"?>
<GetIpamPoolAllocationsResponse xmlns="http://ec2.amazonaws.com/doc/2016-11-15/">
    <requestId>0ddfc599-4746-46ba-bcbe-***</requestId>
    <ipamPoolAllocationSet>
        <item>
            <allocationId>ipam-pool-alloc-***</allocationId>
            <cidr>192.168.1.0/24</cidr>
            <cidrBlock>192.168.1.0/24</cidrBlock>
            <ipamPoolAllocationId>ipam-pool-alloc-***</ipamPoolAllocationId>
            <resourceId>vpc-****</resourceId>
            <resourceOwner>***</resourceOwner>
            <resourceRegion>us-east-2</resourceRegion>
            <resourceType>vpc</resourceType>
        </item>
    </ipamPoolAllocationSet>
</GetIpamPoolAllocationsResponse>" http.response.header.cache_control="no-cache, no-store" http.response.header.server=AmazonEC2 @caller=github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2@v2.0.0-beta.26/logger.go:138 http.response.header.strict_transport_security="max-age=31536000; includeSubDomains" http.status_code=200 aws.region=us-east-2 aws.service=EC2 http.response.header.x_amzn_requestid=0ddfc599-4746-46ba-bcbe-*** tf_req_id=3c02edf2-d892-1132-26f2-dbb1083f9dbe tf_resource_type=aws_vpc tf_rpc=ApplyResourceChange http.response.header.vary=accept-encoding http.duration=332 http.response.header.content_type=text/xml;charset=UTF-8 http.response.header.date="Fri, 05 May 2023 07:41:26 GMT" tf_mux_provider=*schema.GRPCProviderServer tf_provider_addr=registry.terraform.io/hashicorp/aws @module=aws aws.operation=GetIpamPoolAllocations timestamp=2023-05-05T09:41:26.556+0200
2023-05-05T09:41:26.557+0200 [TRACE] provider.terraform-provider-aws_v4.64.0_x5: [TRACE] Waiting 10s before next try
2023-05-05T09:41:29.465+0200 [TRACE] dag/walk: vertex "root" is waiting for "module.production.module.vpc.provider[\"registry.terraform.io/hashicorp/aws\"] (close)"
2023-05-05T09:41:29.465+0200 [TRACE] dag/walk: vertex "module.production.module.vpc.provider[\"registry.terraform.io/hashicorp/aws\"] (close)" is waiting for "module.production.module.vpc.aws_vpc.vpc (destroy)"
2023-05-05T09:41:34.468+0200 [TRACE] dag/walk: vertex "module.production.module.vpc.provider[\"registry.terraform.io/hashicorp/aws\"] (close)" is waiting for "module.production.module.vpc.aws_vpc.vpc (destroy)"
2023-05-05T09:41:34.468+0200 [TRACE] dag/walk: vertex "root" is waiting for "module.production.module.vpc.provider[\"registry.terraform.io/hashicorp/aws\"] (close)"
module.production.module.vpc.aws_vpc.vpc: Still destroying... [id=vpc-****, 13m40s elapsed]

```

Debug when deletion is complete:
```
module.production.module.vpc.aws_vpc.vpc: Still destroying... [id=vpc-***, 13m50s elapsed]
2023-05-05T09:41:46.904+0200 [DEBUG] provider.terraform-provider-aws_v4.64.0_x5: HTTP Request Sent: http.request.header.authorization="AWS4-HMAC-SHA256 Credential=AKIA************BI2I/20230505/us-east-2/ec2/aws4_request, SignedHeaders=content-length;content-type;host;x-amz-date, Signature=*****" http.request.header.x_amz_date=20230505T074146Z http.request_content_length=87 tf_provider_addr=registry.terraform.io/hashicorp/aws tf_resource_type=aws_vpc net.peer.name=ec2.us-east-2.amazonaws.com tf_mux_provider=*schema.GRPCProviderServer http.request.header.content_type="application/x-www-form-urlencoded; charset=utf-8" http.url=https://ec2.us-east-2.amazonaws.com/ http.user_agent="APN/1.0 HashiCorp/1.0 Terraform/1.3.9 (+https://www.terraform.io) terraform-provider-aws/4.64.0 (+https://registry.terraform.io/providers/hashicorp/aws) aws-sdk-go/1.44.246 (go1.19.8; linux; amd64)" @caller=github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2@v2.0.0-beta.26/logger.go:90 @module=aws aws.sdk=aws-sdk-go aws.service=EC2 http.method=POST tf_rpc=ApplyResourceChange aws.operation=GetIpamPoolAllocations aws.region=us-east-2 http.flavor=1.1 http.request.body=Action=GetIpamPoolAllocations&IpamPoolId=ipam-pool-***&Version=2016-11-15 tf_req_id=3c02edf2-d892-1132-26f2-dbb1083f9dbe timestamp=2023-05-05T09:41:46.904+0200
2023-05-05T09:41:47.274+0200 [DEBUG] provider.terraform-provider-aws_v4.64.0_x5: HTTP Response Received: http.duration=369 http.response.body="<?xml version="1.0" encoding="UTF-8"?>
<GetIpamPoolAllocationsResponse xmlns="http://ec2.amazonaws.com/doc/2016-11-15/">
    <requestId>402aeb87-a130-43cd-b282-***</requestId>
    <ipamPoolAllocationSet/>
</GetIpamPoolAllocationsResponse>" http.response.header.cache_control="no-cache, no-store" http.status_code=200 tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2@v2.0.0-beta.26/logger.go:138 aws.region=us-east-2 http.response.header.content_type=text/xml;charset=UTF-8 http.response.header.vary=accept-encoding @module=aws aws.operation=GetIpamPoolAllocations http.response.header.strict_transport_security="max-age=31536000; includeSubDomains" http.response.header.x_amzn_requestid=402aeb87-a130-43cd-b282-*** tf_provider_addr=registry.terraform.io/hashicorp/aws tf_resource_type=aws_vpc aws.service=EC2 http.response.header.date="Fri, 05 May 2023 07:41:46 GMT" tf_req_id=3c02edf2-d892-1132-26f2-dbb1083f9dbe aws.sdk=aws-sdk-go http.response.header.server=AmazonEC2 tf_mux_provider=*schema.GRPCProviderServer timestamp=2023-05-05T09:41:47.274+0200
2023-05-05T09:41:47.274+0200 [TRACE] provider.terraform-provider-aws_v4.64.0_x5: Called downstream: tf_provider_addr=registry.terraform.io/hashicorp/aws tf_resource_type=aws_vpc @caller=github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:807 tf_mux_provider=*schema.GRPCProviderServer tf_req_id=3c02edf2-d892-1132-26f2-dbb1083f9dbe tf_rpc=ApplyResourceChange @module=sdk.helper_schema timestamp=2023-05-05T09:41:47.274+0200
2023-05-05T09:41:47.274+0200 [TRACE] provider.terraform-provider-aws_v4.64.0_x5: Received downstream response: @module=sdk.proto diagnostic_warning_count=0 tf_proto_version=5.3 tf_provider_addr=registry.terraform.io/hashicorp/aws tf_req_duration_ms=831935 tf_req_id=3c02edf2-d892-1132-26f2-dbb1083f9dbe tf_resource_type=aws_vpc @caller=github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tf5serverlogging/downstream_request.go:37 diagnostic_error_count=0 tf_rpc=ApplyResourceChange timestamp=2023-05-05T09:41:47.274+0200
2023-05-05T09:41:47.274+0200 [TRACE] provider.terraform-provider-aws_v4.64.0_x5: Served request: tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:831 @module=sdk.proto tf_proto_version=5.3 tf_provider_addr=registry.terraform.io/hashicorp/aws tf_req_id=3c02edf2-d892-1132-26f2-dbb1083f9dbe tf_resource_type=aws_vpc timestamp=2023-05-05T09:41:47.274+0200
2023-05-05T09:41:47.275+0200 [TRACE] NodeAbstractResouceInstance.writeResourceInstanceState to workingState for module.production.module.vpc.aws_vpc.vpc
2023-05-05T09:41:47.275+0200 [TRACE] NodeAbstractResouceInstance.writeResourceInstanceState: removing state object for module.production.module.vpc.aws_vpc.vpc
module.production.module.vpc.aws_vpc.vpc: Destruction complete after 13m52s
```

### Would you like to implement a fix?

No

**Comment by justinretzolk** [comment]
Hey @TibbeVandenBerghe 👋 Thank you for checking in on this! Unfortunately I can't provide an ETA on when this will be looked into due to the potential of shifting priorities. We prioritize by count of :+1: reactions and a few other things (more information on our [prioritization guide](https://hashicorp.github.io/terraform-provider-aws/prioritization/) if you're interested).

**Comment by mattburgess** [comment]
The wait was introduced in the fix for #27571. I can imagine if there's a force-recreation of a VPC whose CIDR is IPAM-managed then you could end up with this sequence of events:

1. VPC is initially created successfully; an IPAM allocation is created behind the scenes
2. VPC is deleted due to a change in tenancy, for example, and this deletion succeeds in a few seconds
3. An attempt to create the replacement VPC is made, but will fail because the VPC's CIDR is still linked to an IPAM allocation which is itself linked to the now deleted VPC.

Having tested the same sequence of events in the AWS console, the slowness of releasing that IPAM allocation is reproducible there so I'm not sure there's much we can do without sacrificing correctness. It might be worth reaching out to your AWS TAM/AWS Support to see if they have any suggestions on potential workarounds, or if there are plans in the works to reduce the length of time those allocation releases take.
